internal/manifest/constants.go

```diff
@@ -3,32 +3,37 @@
 
 package manifest
 
-// TrustedMeasurement contains the expected launch digest and is injected at build time.
-var TrustedMeasurement = "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+import (
+	"encoding/json"
+	"fmt"
+	"os"
 
-// Default returns a default manifest.
-func Default() Manifest {
-	return Manifest{
-		ReferenceValues: ReferenceValues{
-			TrustedMeasurement: HexString(TrustedMeasurement),
-		},
+	"github.com/edgelesssys/contrast/node-installer/platforms"
+)
+
+// Default returns a default manifest with reference values for the given platform.
+func Default(platform platforms.Platform) *Manifest {
+	if embeddedReferenceValues == nil {
+		// If we're here, this is the first time this function is called, and the global state is not
+		// yet initialized. So let's unmarshal the embedded reference values. This does not need to be
+		// locked, as embeddedReferenceValues has a fixed value at compile time, making this idempotent.
+		if err := json.Unmarshal(EmbeddedReferenceValuesJSON, &embeddedReferenceValues); err != nil {
+			fmt.Printf("Failed to unmarshal embedded reference values: %s\n", err)
+			os.Exit(1)
+		}
 	}
-}
 
-// DefaultAKS returns a default manifest with AKS reference values.
-func DefaultAKS() Manifest {
-	mnfst := Default()
-	mnfst.ReferenceValues.SNP = SNPReferenceValues{
-		MinimumTCB: SNPTCB{
-			BootloaderVersion: toPtr(SVN(3)),
-			TEEVersion:        toPtr(SVN(0)),
-			SNPVersion:        toPtr(SVN(8)),
-			MicrocodeVersion:  toPtr(SVN(115)),
-		},
+	mnfst := Manifest{}
+	switch platform {
+	case platforms.AKSCloudHypervisorSNP:
+		mnfst.ReferenceValues.AKS = embeddedReferenceValues.AKS
+	case platforms.RKE2QEMUTDX, platforms.K3sQEMUTDX:
+		mnfst.ReferenceValues.BareMetalTDX = embeddedReferenceValues.BareMetalTDX
 	}
-	return mnfst
+	return &mnfst
 }
 
-func toPtr[T any](t T) *T {
-	return &t
+// DefaultPlatformHandler is a short-hand for getting the default runtime handler for a platform.
+func DefaultPlatformHandler(platform platforms.Platform) (string, error) {
+	return Default(platform).RuntimeHandler(platform)
```

Comment:
This looks odd - why does the `RuntimeHandler` func need a platform argument?

Reply:
Because the runtime-handler is platform-specific (as its name is derived from the launch measurement)

Reply:
You mean the manifest could contain multiple sets of reference values, so it can't decide for which one to return the RuntimeHandler?

Reply:
I think it was the plan to allow this, yes. (We'll also have a problem once we have multiple allowed trusted measurements for one platform though. No clue how to handle that)